### DISCUSSION TITLE
[V7] Select rows and Server side pagination

### DISCUSSION THREAD

**Discussion by 07harish** [TanStack/table#2351] [General] [answered]
**TL'DR:  In server side pagination, When I select any rows in 1st page using the checkbox, those rows show selected in all pages. I am guessing it is based on **row ID**, so it shows as selected.**

**For example: If I select 3rd & 4th row in 1st page, then I move to other pages, I see 3rd & 4th row as selected in all of the pages as row ID starts from zero in all pages**

 I am implementing server-side pagination & the table also has a checkbox column for selecting rows. 

Pagination is little different - I fetch 200 records & display 40rows per page. In the 5th page, I fetch 200 more records & keep displaying 40rows per page... this continues till the end where I fetch 200 & pageSize will be 40 & I am able to implement this.

When I log selectedFlatRows - Rows are selected correctly, but on UI the rows are selected based on **row ID** &  also **on all pages index & row id starts from zero**. hence row is shown selected on all of the pages. 

for example :When I select 0th rowID in 1st page, I get that row selected in selectFlatRows correctly, But when I move to other pages & deselect it, it gets removed.

Please help me in solving this. Thanks for reading.

**Comment by justin-tufte** [comment, marked as answer]
This should be the same solution I commented on here:
https://github.com/tannerlinsley/react-table/discussions/2037#discussioncomment-14068

**Reply by 07harish** [reply]
Thank you so much. It looks exactly what I wanted. Since it is based on row ID. I am trying this.

**Reply by 07harish** [reply]
Hey @justin-tufte, Thank you. It works. Now I can give unique row ID. 

I am using server-side pagination, I have logged selectedFlatRows. Now when I select rows in the 1st page I get that logged, when I move to the 2nd page, It gets cleared &I see the empty array in the console. Again when I come back to the 1st page, I get my selected rows.

Not sure how it holds the state of 1st page & 2nd page has its own state with empty array (selected flat rows)

Do you know any solution for this, please?
